### PR TITLE
support for nb7

### LINKED ISSUE #167: Extend support to notebook 7
Jupyter Notebook 7 is about to be released (as of writing, b3 released).

In this new version, most of the backend is based on Jupyter server. While https://github.com/danielfrg/s3contents/issues/161 has solved some compatibility issues, some problem remains.

For instance, there are still some `notebook.*` imports in `ipycompat.py` that raise exceptions on startup:

```
[E 2023-06-09 09:05:17.328 JupyterNotebookApp] Exception while loading config file /home/jupyter/.jupyter/jupyter_notebook_config.py
    Traceback (most recent call last):
      File "/home/jupyter/.local/lib/python3.10/site-packages/traitlets/config/application.py", line 909, in _load_config_files
        config = loader.load_config()
      File "/home/jupyter/.local/lib/python3.10/site-packages/traitlets/config/loader.py", line 626, in load_config
        self._read_file_as_dict()
      File "/home/jupyter/.local/lib/python3.10/site-packages/traitlets/config/loader.py", line 659, in _read_file_as_dict
        exec(compile(f.read(), conf_filename, "exec"), namespace, namespace)  # noqa
      File "/home/jupyter/.jupyter/jupyter_notebook_config.py", line 2, in <module>
        from s3contents import S3ContentsManager
      File "/home/jupyter/.local/lib/python3.10/site-packages/s3contents/__init__.py", line 3, in <module>
        from .s3manager import S3ContentsManager  # noqa
      File "/home/jupyter/.local/lib/python3.10/site-packages/s3contents/s3manager.py", line 6, in <module>
        from s3contents.genericmanager import GenericContentsManager, from_dict
      File "/home/jupyter/.local/lib/python3.10/site-packages/s3contents/genericmanager.py", line 17, in <module>
        from s3contents.genericfs import GenericFSError, NoSuchFile
      File "/home/jupyter/.local/lib/python3.10/site-packages/s3contents/genericfs.py", line 5, in <module>
        from s3contents.ipycompat import HasTraits
      File "/home/jupyter/.local/lib/python3.10/site-packages/s3contents/ipycompat.py", line 18, in <module>
        from notebook.services.contents.checkpoints import (
    ModuleNotFoundError: No module named 'notebook.services'
```

I will PR a change that I've just tested to make it compatible with Notebook 7.

### PR DESCRIPTION
**Pull request by antoine-galataud**
Fixes #167

### CODE REVIEW

File: s3contents/ipycompat.py [hunk 1]
```diff
@@ -15,28 +14,40 @@
...
+    except ModuleNotFoundError:
+        pass
+
+if not ct_mgr_deps_loaded:
+    raise ImportError(
+        "Couldn't import ContentsManager from either nootebook or jupyter_server."
+        "Make sure that Jupyter Notebook or JupyterLab package is installed."
+    )
 
-from notebook.utils import to_os_path
```
Comment (antoine-galataud) [inline s3contents/ipycompat.py:39]: this method wasn't used anywhere

**Comment by danielfrg** [comment]
Thanks for the PR. Could you add notebook 7 to the test matrix?

**Comment by antoine-galataud** [comment]
@danielfrg sorry, very late answer. I just added it to the test matrix.

**Comment by antoine-galataud** [comment]
Hum of course, 7.0 isn’t released yet. I’ll just make sure it passes with rc2 already.

**Comment by danielfrg** [comment]
Thank you!

**Comment by antoine-galataud** [comment]
notebook 7.0 was released last week, I've bumped version in GH action file.

**Comment by danielfrg** [comment]
Thanks for handling this! I will try to make a release later today.

**Comment by danielfrg** [comment]
A new version with this is on PyPI!

**Pull request merged**